File: java/lib/src/main/java/com/svix/internalapi/Endpoint.java
// this file is @generated
package com.svix.internalapi;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.EndpointTransformationIn;

import lombok.Getter;

import okhttp3.HttpUrl;

import java.io.IOException;

public class Endpoint {
    private final SvixHttpClient client;

    @Getter private final EndpointAutoConfig autoConfig;

    public Endpoint(SvixHttpClient client) {
        this.client = client;
        this.autoConfig = new EndpointAutoConfig(client);
    }

    /**
     * This operation was renamed to `set-transformation`.
     *
     * @deprecated
     */
    @Deprecated
    public void transformationPartialUpdate(
            final String appId,
            final String endpointId,
            final EndpointTransformationIn endpointTransformationIn)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/app/%s/endpoint/%s/transformation",
                                        appId, endpointId));
        this.client.executeRequest("PATCH", url.build(), null, endpointTransformationIn, null);
    }
}

File: kotlin/lib/src/main/kotlin/internalapi/Endpoint.kt
// this file is @generated
package com.svix.kotlin.internal

import com.svix.kotlin.SvixHttpClient
import com.svix.kotlin.models.EndpointTransformationIn

class Endpoint(private val client: SvixHttpClient) {
    val autoConfig: EndpointAutoConfig = EndpointAutoConfig(client)

    /**
     * This operation was renamed to `set-transformation`.
     *
     * @deprecated
     */
    @Deprecated("")
    suspend fun transformationPartialUpdate(
        appId: String,
        endpointId: String,
        endpointTransformationIn: EndpointTransformationIn,
    ) {
        val url =
            client
                .newUrlBuilder()
                .encodedPath("/api/v1/app/$appId/endpoint/$endpointId/transformation")

        client.executeRequest<EndpointTransformationIn, Boolean>(
            "PATCH",
            url.build(),
            reqBody = endpointTransformationIn,
        )
    }
}
